tools/code-generation/generator/src/main/resources/com/amazonaws/util/awsclientgenerator/velocity/cpp/json/JsonErrorMarshallerHeader.vm

```diff
@@ -21,6 +21,9 @@ namespace ${serviceNamespace}
 class ${CppViewHelper.computeExportValue($metadata.classNamePrefix)} $className : public Aws::Client::JsonErrorMarshaller
 {
 public:
+#if ($metadata.awsQueryCompatible)
+  explicit $className(bool queryCompatibilityMode) : Aws::Client::JsonErrorMarshaller(queryCompatibilityMode){}
```

Comment:
Two comments:

first:

you need `:public Aws::Client::JsonErrorMarshaller` otherwise i dont think this would actually work. i.e. in the [aws client constuctor](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/client/AWSJsonClient.h#L40) 

```cpp
AWSJsonClient(const Aws::Client::ClientConfiguration& configuration,
  const std::shared_ptr<Aws::Client::AWSAuthSigner>& signer,
  const std::shared_ptr<AWSErrorMarshaller>& errorMarshaller);
```

since we are missing `public` and `is a` inheritance, i dont think this would end up compiling.

second:

I think that we can make this code a little bit more maintainable using a decorator instead of a boolean. This also forces the query compatible decision to compile time and not runtime. i.e. we could do

```cpp
class JsonErrorMarshallerQueryCompatible: public JsonErrorMarshaller {
  public:
    AWSError<CoreErrors> Marshall(const Aws::Http::HttpResponse& response) const override {
       //do special query compatible stuff
    }

AWSError<CoreErrors> BuildAWSError(const std::shared_ptr<Http::HttpResponse>& httpResponse) const override {
       //do special query compatible stuff
    }
};
```

then at code generation time we could do 

```
  explicit $className : public Aws::Client::JsonErrorMarshallerQueryCompatible
```

and avoid passing around true and falses